waspc/data/Generator/templates/sdk/wasp/env/index.ts

```diff
@@ -0,0 +1,29 @@
+import * as z from 'zod'
+
+const redColor = '\x1b[31m'
+
+export function ensureEnvSchema<Schema extends z.ZodTypeAny>(
+  data: unknown,
+  schema: Schema
+): z.infer<Schema> {
+  try {
+    return schema.parse(data)
+  } catch (e) {
+    // TODO: figure out how to output the error message in a better way
```

Comment:
| ![Screenshot 2024-11-29 at 09 04 34](https://github.com/user-attachments/assets/752f487c-9fc7-43a9-8a5a-eae64af7103c) |
| --- |
| How it looks on the server |

| ![Screenshot 2024-11-29 at 09 14 05](https://github.com/user-attachments/assets/3bb64f92-7957-4a1a-b203-6c12d834ba5d) |
| --- |
| Error on the client |

I'm not sure if this is good enough?

Reply:
I would probably experiment with the title not being a box, but just being underlined title. When it is a box it feels like it is a standalone thing, that stuff under it does not belong to it. Also, indent the bullet points, to make it even clearer they are "under" it.

So
```
===== Env vars validation failed =====
 - REACT_APP_WS_URL is required
 - SOME_VAR is required
======================================
```

Reply:
This screenshots are old news :) the code has changed and it doesn't output a box anymore.